lib/main.dart

```diff
@@ -1,20 +1,19 @@
 import 'package:flutter/material.dart';
-import 'package:wh_covid19/routes.dart';
-import 'package:wh_covid19/hard_data.dart';
-import 'package:wh_covid19/style.dart';
-import 'package:wh_covid19/view/home_page.dart';
-import 'package:wh_covid19/view/icu_non_intensivist/general_care_view.dart';
-import 'package:wh_covid19/view/icu_non_intensivist/tips_junior_staff_view.dart';
-import 'package:wh_covid19/view/icu_non_intensivist/ventilation/ventilation_view.dart';
-import 'package:wh_covid19/view/info_view.dart';
-import 'package:wh_covid19/view/intubation/algorithm/intubation_algorithm_page.dart';
-import 'package:wh_covid19/view/intubation/checklist/intubation_checklist_page.dart';
-import 'package:wh_covid19/view/intubation/guidance/intubation_guidance_page.dart';
-import 'package:wh_covid19/view/ppe/ppe_view.dart';
-import 'package:wh_covid19/view/sbs_guide_view.dart';
-import 'package:wh_covid19/view/view_templates/html_text_card_view_template.dart';
-import 'package:wh_covid19/view/staff_welfare/your_welfare_view.dart';
-
+import 'hard_data.dart';
```

Comment:
@domesticmouse I thought the recommendation was to use package imports rather than relative imports?

Reply:
Uh, where did you get that from? (It actually leads to the type system getting confused, because it thinks the package local type and the external package import are two different types.)

Reply:
Interesting! Not sure I read it anywhere, but I personally have preferred it for consistency.

Also:
https://github.com/flutter/flutter/blob/86389be66e8e0bd3e082b20a6bb3251d7c7706d5/analysis_options.yaml#L73
🤷

Although it does say to prefer relative here:
https://dart.dev/guides/language/effective-dart/usage#prefer-relative-paths-when-importing-libraries-within-your-own-packages-lib-directory
But then it says: "There is no profound reason to prefer the former—it’s just shorter, and we want to be consistent."

But then it does say:
An import path should never contain /lib/
A library under lib should never use ../ to escape the lib directory.

So I guess relative path like "view/home_page.dart" is recommended, but a relative path like "import '../hard_data.dart';" is not recommended. At least according to the docs 🤔

Reply:
It's something I'm guessing Java/Android developers will do because it feels right. 

For better or worse, in Dart path relative imports and package imports mean different things to the Dart's type system. Thus the lint checks =)

Reply:
> It's something I'm guessing Java/Android developers will do because it feels right.

That explains why I have a leaning toward it :)

This PR still has '../' style imports so I think those should be removed as per the dart.dev guide?

Reply:
The `../` style imports are not escaping the `lib` directory. They are just traversing within the `lib` directory, and thus are totes kosher.

Reply:
Ah ok got it. Thanks for the clarification.

Reply:
Also, if you're saying there is a "real" reason to use relative versus package, you may want to ask them to update the docs which say:
"There is no profound reason to prefer the former—it’s just shorter, and we want to be consistent."

Reply:
Also:
https://github.com/dart-lang/pedantic/blob/master/lib/analysis_options.1.9.0.yaml#L16

Reply:
Yup, I've got that lint included as well: 

https://github.com/domesticmouse/wh_covid19_app/blob/Adding-code-style-checks/analysis_options.yaml#L28

Reply:
Actually, I should be getting a warning that it is double included. Meh.

Reply:
@domesticmouse it seems to be used all over the place in Googles own samples:

https://github.com/flutter/samples/tree/master/veggieseasons/lib/main.dart
https://github.com/flutter/gallery/blob/master/lib/main.dart
https://github.com/flutter/samples/blob/master/provider_shopper/lib/main.dart
etc

but I think this maybe due a old bug since fixed as most recent comments on Remi's answer: https://stackoverflow.com/questions/47121411/flutter-retrieving-top-level-state-from-child-returns-null/47142052#47142052